src/inverse-kinematics/include/iDynTree/InverseKinematics.h

```diff
@@ -705,6 +705,16 @@ class iDynTree::InverseKinematics
     
     void deactivateCOMTarget();
 
+    /*!
+     * Set the directions along which a point will be projected.
+     *
+     * \author Aiko Dinale (29/08/2017)
+     *
+     * @param direction     vector along which we want to project a point
+     */
+    void setProjectionDirection(iDynTree::Vector3 direction,
+                                const iDynTree::Position originOfPlaneInWorld = iDynTree::Position::Zero());
```

Comment:
We are modifying the origin set in the ConvexHull here. We should keep the origin that is set while the convex hull is built.

Reply:
The name of the method is a bit confusing. I would suggest `setCOMConstraintProjectionDirection` to clarify the projection that we are referencing.

Reply:
You're definitely right, no idea of why I did so. Now the method is simply:
```C++
void setCOMConstraintProjectionDirection(iDynTree::Vector3 direction);
```